src/python_framework_v02/troute/nhd_io.py

```diff
@@ -421,32 +510,42 @@ def get_usgs_from_time_slices_folder(
     usgs_df = usgs_df.drop(["gages", "ascendingIndex", "to"], axis=1)
     columns_list = usgs_df.columns
 
-    for i in range(0, (len(columns_list) * 3) - 12, 12):
-        original_string = usgs_df.columns[i]
-        original_string_shortened = original_string[:-5]
-        temp_name1 = original_string_shortened + str("05:00")
-        temp_name2 = original_string_shortened + str("10:00")
-        temp_name3 = original_string_shortened + str("20:00")
-        temp_name4 = original_string_shortened + str("25:00")
-        temp_name5 = original_string_shortened + str("35:00")
-        temp_name6 = original_string_shortened + str("40:00")
-        temp_name7 = original_string_shortened + str("50:00")
-        temp_name8 = original_string_shortened + str("55:00")
-        usgs_df.insert(i + 1, temp_name1, np.nan)
-        usgs_df.insert(i + 2, temp_name2, np.nan)
-        usgs_df.insert(i + 4, temp_name3, np.nan)
-        usgs_df.insert(i + 5, temp_name4, np.nan)
-        usgs_df.insert(i + 7, temp_name5, np.nan)
-        usgs_df.insert(i + 8, temp_name6, np.nan)
-        usgs_df.insert(i + 10, temp_name7, np.nan)
-        usgs_df.insert(i + 11, temp_name8, np.nan)
+    original_string_first = usgs_df.columns[0]
+    date_time_str = original_string_first[:10] + " " + original_string_first[11:]
+    date_time_obj_start = datetime.strptime(date_time_str, "%Y-%m-%d %H:%M:%S")
+
+    original_string_last = usgs_df.columns[-1]
+    date_time_str = original_string_last[:10] + " " + original_string_last[11:]
+    date_time_obj_end = datetime.strptime(date_time_str, "%Y-%m-%d %H:%M:%S")
+
+    def daterange(start, end, step=timedelta(1 / 288)):
+        print(start, end)
+        curr = start
+        while curr < end:
+            yield curr
+            curr += step
+
+    dates = []
+    for j in daterange(date_time_obj_start, date_time_obj_end):
+        dates.append(j.strftime("%Y-%m-%d_%H:%M:00"))
+
+    for i, j in enumerate(dates):
+        if usgs_df.iloc[:, i].name == dates[i]:
+            pass
+        else:
+            if usgs_df.iloc[:, i].name not in dates:
+                usgs_df.drop(usgs_df.columns[i], axis=1, inplace=True)
```

Comment:
ping @groutr. We discovered that some of the time stamps in the USGS time slices files are unexpected -- there are some that fall on 5 minute marks, which is sort-of ok, and there are some that fall on sub-five-minute intervals. As we explored the datasets, it appears that the data in the for example 15:31 timestamp is identical to the 15:30 timestamp. SO... we are deleting what appear to be duplicated entries and interpolating over them.